code/modules/atmospherics/gasmixtures/gas_mixture.dm

```diff
@@ -192,6 +192,23 @@ GLOBAL_LIST_INIT(gaslist_cache, init_gaslist_cache())
 
 	return removed
 
+	///Removes a amount of a specific gas from the gas_mixture.
+	///Returns: gas_mixture with the gas removed
+/datum/gas_mixture/proc/remove_specific(gas_id, amount)
+	var/list/cached_gases = gases
+	amount = min(amount, cached_gases[gas_id][MOLES])
+	if(amount <= 0)
+		return null
+	var/datum/gas_mixture/removed = new type
+	var/list/removed_gases = removed.gases
+	removed.temperature = temperature
+	ADD_GAS(gas_id, removed.gases)
+	removed_gases[gas_id][MOLES] = amount
+	cached_gases[gas_id][MOLES] -= amount
+
+	garbage_collect(list(gas_id))
+	return removed
+

```

Comment:
is this being used?

Reply:
Within the proc yes, but i'm not using that proc anymore.  The proc is left in because Dennok said it would be a nice-to-have.

Reply:
This proc can be used in air filters 
and in any place where you need take only one gas from mixture,
and dont want to copypaste parts of code,
and dont boring about absolute clearness.